relay/README.md

```diff
@@ -86,93 +64,144 @@ Scene 2:
+- `/p2p-circuit/ip4/../tcp/../p2p/QmTwo`
+  - Dial QmTwo, through any available relay node, but force the relay node to use the encapsulated `/ip4` multiaddr for connecting to QmTwo.
+
+Specify a relay:
+
 - `/p2p/QmRelay/p2p-circuit/p2p/QmTwo`
   - Dial QmTwo, through QmRelay.
   - Use peer routing to find an address for QmRelay.
   - The relay node will also use peer routing, to find an address for QmTwo.
-- `/p2p-circuit/ip4/../tcp/../p2p/QmTwo`
-  - Dial QmTwo, through any available relay node,
-    but force the relay node to use the encapsulated `/ip4` multiaddr for connecting to QmTwo.
-  - We'll probably not support forced addresses for now, just because it's complicated.
-- `/ip4/../tcp/../p2p/QmRelay/p2p-circuit`
-  - Listen for connections relayed through QmRelay.
-  - Includes info for connecting to QmRelay.
-  - Also makes QmRelay available for relayed dialing, based on how listeners currently relate to dialers.
-- `/p2p/QmRelay/p2p-circuit`
-  - Same as previous example, but use peer routing to find an address for QmRelay.
-- `/p2p-circuit/p2p/QmTwo/p2p-circuit/p2p/QmThree`
-  - Dial QmThree, through a relayed connection to QmTwo.
-  - The relay nodes will use peer routing to find an address for QmTwo and QmThree.
-  - We'll probably not support nested relayed connections for now, there are edge cases to think of.
 - `/ip4/../tcp/../p2p/QmRelay/p2p-circuit/p2p/QmTwo`
   - Dial QmTwo, through QmRelay.
   - Includes info for connecting to QmRelay.
   - The relay node will use peer routing to find an address for QmTwo.
-- `/p2p-circuit`
-  - Use relay discovery to find a suitable relay node. (Neither specified nor implemented.)
-  - Listen for relayed connections.
-  - Dial through the discovered relay node for any `/p2p-circuit` multiaddr.
 
-TODO: figure out forced addresses.
-TODO: figure out nested relayed connections.
+Double relay:
+
+- `/p2p-circuit/p2p/QmTwo/p2p-circuit/p2p/QmThree`
+  - Dial QmThree, through a relayed connection to QmTwo.
+  - The relay nodes will use peer routing to find an address for QmTwo and QmThree.
+  - We'll **not support nested relayed connections for now**, see [Future Work](#future-work) section.
 
 ## Wire format
 
-The wire format (or codec) is named `/ipfs/relay/circuit` and is simple.
-A variable-length header consisting of two length-prefixed multiaddrs
-is followed by a bidirectional stream of arbitrary data,
-and the eventual closing of the stream.
+We start the description of the Wire format by illustrating a possible flow scenario and then describing them in detail by phases.
+
+### Relay Message
+
+Every message in the relay protocol uses the following protobuf:
 
 ```
-<src><dst><data>
-
- ^    ^    ^
- |    |    |
- |    |    +-- bidirectional data stream
- |    |         (usually /multistream-select in the case of /p2p multiaddrs)
- |    |
- |    +------- multiaddr of the listening node
- |
- +------------ multiaddr of the dialing node
+message CircuitRelay {
```

Comment:
Looks good, but should we have a version as part of the message as well? I know its communicated as part of the multistream-select rpc end point, but having a version in the message might help preventing stupid mistakes like `we made changes to the protobuff msg, but forgot to bump the multistream version`.